src/utils/common-utils.ts

```diff
@@ -232,3 +232,27 @@ export function calcAccuracyMultiplier(userAccStage: number, targetEvaStage: num
 
   return 1;
 }
+
+/**
+ * Returns the name of the key that matches the enum value.
+ *
+ * **Note**: **DO NOT** use this with TypeScript enums, only with `const` objects being used as enums.
+ * @param input - The enum to check
+ * @param val - The value to get the key of
+ * @returns The name of the key with the specified value
+ * @example
+ * const thing = {
+ *   one: 1,
+ *   two: 2,
+ * } as const;
+ * console.log(enumValueToKey(thing, thing.two)); // output: "two"
+ * @throws An `Error` if an invalid enum value is passed to the function
+ */
+export function enumValueToKey(input: object, val: number | string): string | never {
+  for (const [key, value] of Object.entries(input)) {
+    if (val === value) {
+      return key;
+    }
+  }
+  throw new Error(`Invalid value passed to \`enumValueToKey\`! Value: ${val}`);
+}
```

Comment:
You should auto-inject the type through generic tpying to persist the actual type of return

```suggestion
export function enumValueToKey<T extends Record<string, string | number>>(input: T, val: T[keyof T]): keyof T {
  for (const [key, value] of Object.entries(input)) {
    if (val === value) {
      return key as keyof T;
    }
  }
  throw new Error(`Invalid value passed to \`enumValueToKey\`! Value: ${val}`);
}
```

| Before | After |
|--------|--------|
| <img width="490" alt="image" src="https://github.com/user-attachments/assets/c78b709c-f1b4-42ab-bf97-9a9022862e0d" /> | <img width="698" alt="image" src="https://github.com/user-attachments/assets/c65c1fe5-8c37-4ebe-902b-d1c3c4d9cbf2" /> |

Reply:
Ah, good idea.

Reply:
Feel free to experiment with _better_ or different approaches but you get the goal of it